app/scripts/lib/inpage-provider.js

```diff
@@ -27,14 +29,24 @@ function MetamaskInpageProvider (connectionStream) {
   )
 
   // connect to async provider
-  const asyncProvider = self.asyncProvider = new StreamProvider()
+  const engine = new ProviderEngine()
+
+  const filterSubprovider = new FilterSubprovider()
+  engine.addProvider(filterSubprovider)
+
+  const stream = self.stream = new StreamSubprovider()
```

Comment:
Like `RemoteStreamSubprovider`?  I'm not sure what you mean less generic. Isn't descriptive good here?